File: web/src/types/audit.ts
export interface AuditEvent {
  audit_event_id?: string
  event_id?: string
  request_id: string
  decision_id?: string
  workflow_type?: string
  node_name?: string
  actor: string
  action: string
  node_action?: string
  proposal_id?: string
  confirmation_id?: string
  error_case_id?: string
  status: string
  error_code?: string
  before_state?: string
  after_state?: string
  rule_version?: string
  snapshot_id?: string
  input_ref_type?: string
  input_ref?: string
  output_ref_type?: string
  output_ref?: string
  created_at: string
}

File: internal/application/dto/audit.go
package dto

// AuditEventDTO 是审计时间线展示结构。
// 字段与 audit_events 表保持对应，便于前端追踪请求、决策、规则和错误案例。
type AuditEventDTO struct {
	AuditEventID   string `json:"audit_event_id"`
	EventID        string `json:"event_id"`
	RequestID      string `json:"request_id"`
	DecisionID     string `json:"decision_id,omitempty"`
	WorkflowType   string `json:"workflow_type,omitempty"`
	NodeName       string `json:"node_name,omitempty"`
	Actor          string `json:"actor"`
	Action         string `json:"action"`
	NodeAction     string `json:"node_action,omitempty"`
	ProposalID     string `json:"proposal_id,omitempty"`
	ConfirmationID string `json:"confirmation_id,omitempty"`
	ErrorCaseID    string `json:"error_case_id,omitempty"`
	Status         string `json:"status"`
	ErrorCode      string `json:"error_code,omitempty"`
	BeforeState    string `json:"before_state,omitempty"`
	AfterState     string `json:"after_state,omitempty"`
	RuleVersion    string `json:"rule_version,omitempty"`
	SnapshotID     string `json:"snapshot_id,omitempty"`
	InputRefType   string `json:"input_ref_type,omitempty"`
	InputRef       string `json:"input_ref,omitempty"`
	OutputRefType  string `json:"output_ref_type,omitempty"`
	OutputRef      string `json:"output_ref,omitempty"`
	CreatedAt      string `json:"created_at"`
}
